### PR TITLE
Start of i18n context implementation

### PR DESCRIPTION
**Pull request by timcowlishaw**
Hi there folks - here's the outline of a Context / Provider pattern implementation of the i18n stuff - I've edited the 'Groups' page component to show how it could be used in practice.

One neat thing about this approach over the library (thanks, @Pingid for the idea) is that referencing UI copy is now typesafe (ie if you try and reference copy that's not written we'll get a type error), and once we add translations, checking them for completeness is also handled by the type checker, which is pretty cool.

I'd be keen on any feedback on my approach - next steps are probably to integrate use of the i18n context for copy elsewhere on the site, create a Spanish translation, and then to look into producing a UI component for a language switcher (I'm guessing this'd roughly follow the pattern set out in `MapProvider.tsx` for updates to the MapContext from the map controls), if that makes sense.

cheers!

Tim

### CODE REVIEW

File: client/src/components/NewLayout/SearchBox.tsx [hunk 1]
```diff
@@ -5,8 +5,10 @@ import { usePlaceMethod, usePlaceState } from '../../contexts/StateContext'
 import icons from '../../utils/icons'
 import { InputGroup } from '../../styles/styles'
 import { MOON_BLUE } from '../../utils/CONSTANTS'
+import { useI18n } from '../../contexts/I18nProvider'
 
 const SearchBox = () => {
+  const t = useI18n().translation.components.search_box
```
Comment (Pingid) [inline client/src/components/NewLayout/SearchBox.tsx:11]: Styling wise I would  prefer ```useI18n(x => x.translation.components.search_box)```. There isn't a good practicle  reason in this cirmunstace but does make it easier to build functional abstractions.

**Comment by timcowlishaw** [comment]
aha that's a nice idea - i'll correct!

**Comment by timcowlishaw** [comment]
Hey folks - just made that change you suggested, @Pingid, and rebased from the current staging. I need to move a bit more copy into the locale, will finish that in the AM. the 'about' page might take a bit of thought as it includes more markup, but will have a think and let you know the options! Hopefully can get it wrapped up and merged tomorrow

**Comment by timcowlishaw** [comment]
Right folks, I think this is ready for merge if you're happy. I'm not 100% sold on the solution to translate the 'about' page (providing a pre-rendered component from the context per-language), but it's the best i can come up with for now. If you've got any better ideas i'm all ears, otherwise can look at refactoring after we've added a translation or two?

**Pull request merged**